administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,40 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		if (!JSession::checkToken('get'))
+		{
+			$this->app->enqueueMessage(JText::_('JINVALID_TOKEN'));
+			$this->app->redirect('index.php');
+		}
+
+		$title = $this->input->getArray(array('jform' => array('title' => 'string')));
+
+		// Get the model
+		$model = $this->getModel();
+		$return = $model->shareToken($title);
+		echo new JResponseJson($return);
+
+		if ($return)
+		{
+			JText::_('COM_CONTENT_TOKEN_SAVED');
+		}
+		else
+		{
+			JText::_('COM_CONTENT_TOKEN_ERROR');
+		}
```

Comment:
This if / else is still useless :-)

You need to output your shared token here, maybe with a status code. So this is your target json:

`{"status": "ok", "token": "aasdfasdfsadf"}`

To achieve that you have to update your `$model->shareToken` method. For the status you can just check if the token is not null / empty etc for the beginning.

```php
$result = array(
    'status' => $status,
    'token' => $token
);
```

`echo new JResponseJson($return);` // Or json_encode($return);